rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,524 @@
+        event = HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            timestamp=ts_sec_to_ms(transaction.timestamp),
+            sequence_index=self.base_tools.get_next_sequence_counter(),
+            location=Location.BLOCKCHAIN,
+            asset=self.eth,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            balance=Balance(),
+            location_label=cdp_creator,
+            counterparty=CPT_SAI,
+            notes=f'Create CDP {cdp_id}',
+        )
+        return event, []
+
+    def _decode_makerdao_close_cdp(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],  # pylint: disable=unused-argument
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        cdp_creator = hex_or_bytes_to_address(tx_log.topics[1])
+        cdp_id = hex_or_bytes_to_int(tx_log.data[128:])
+
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.INFORMATIONAL and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == ZERO and
+                event.counterparty == CPT_SAI and
+                event.notes is not None and event.notes.startswith('Close CDP') is True
+            ):
+                return None, []
+
+        event = HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            timestamp=ts_sec_to_ms(transaction.timestamp),
+            sequence_index=tx_log.log_index,
+            location=Location.BLOCKCHAIN,
+            asset=self.eth,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            balance=Balance(),
+            location_label=cdp_creator,
+            counterparty=CPT_SAI,
+            notes=f'Close CDP {cdp_id}',
+        )
+        return event, []
+
+    def _decode_makerdao_borrow_sai_event(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        cdp_id = hex_or_bytes_to_int(tx_log.topics[2])
+        withdrawer = hex_or_bytes_to_address(tx_log.topics[1])
+        amount_withdrawn_raw = hex_or_bytes_to_int(tx_log.topics[3])
+        amount_withdrawn = asset_normalized_value(amount=amount_withdrawn_raw, asset=self.sai)
+
+        already_decoded = False
+        for _event in decoded_events:
```

Comment:
Hm, why to start the name from an underscore? Afaik names should start from an underscore only if the variable is unused which doesn't seem to be the case here

Reply:
Ohhh in this case, the only reason for using an underscore is to provide a clear separation from the `event` variable defined later.

Totally ignored the convention here. Will fix